app/src/main/java/org/wikipedia/page/PageTitle.java

```diff
@@ -232,7 +232,8 @@ public void setText(@NonNull String convertedFromText) {
     }
 
     @NonNull public String getDisplayText() {
-        return displayText == null ? StringUtil.removeUnderscores(getPrefixedText()) : displayText;
+        return displayText == null ? StringUtil.removeUnderscores(getPrefixedText())
+                : namespace == null ? displayText : StringUtil.addUnderscores(namespace) + ":" + displayText;
     }
 
     public void setDisplayText(@Nullable String displayText) {
```

Comment:
If the `getDisplayText()` function is automatically adding the namespace, then the `setDisplayText()` function should automatically _remove_ the namespace.  We should do something like:
```
this.displayText = namespace == null ? displayText : StringUtil.removeNamespace(displayText);
```

Reply:
Done!